src/actors/context-menus/characteristic-context-menu.ts

```diff
@@ -84,7 +84,13 @@ export const characteristicMenuOptions = (
       const characteristic = actor.system.characteristics[charName];
       (characteristic as any).name = charName; // TODO adding extra properties that's not on type Characteristic
       const speakerName = token?.name ?? actor.prototypeToken.name ?? "";
-      showImproveCharacteristicDialog(actor, "characteristic", characteristic, speakerName);
+      if (
+        characteristic != null &&
+        characteristic.value != null &&
+        Number.isNumeric(characteristic.value)
+      ) {
```

Comment:
I think the check for if there is a formula should be in the `condition` part so that you don't even see the context menu alternative if you don't have a value and formula.

Also I don't think it's necessary to check if characteristic is null since it should always be there.

This also applies to the `Initialize` & `Initialize All` context menus as well (except that you don't need a value for those menu items).

Reply:
I left this check here, but did it again in the condition for the "Improve Characteristic" context menu item.

I also did a similar check in the condition for "Initialize Characteristic" so that context menu item only displays if the characteristic has a valid formula.